docs/migration/v12.md

```diff
@@ -0,0 +1,277 @@
+# Platform v12 migration guide: Vite & React 18
```

Comment:
it's a great guide @KaiVandivier - two comments:

- Shall we have a mention for the changes to better supprot TypeScript?
- Can we already have this merged with a disclaimer that it is still in alpha, so that people can give it a go?

Reply:
1. Definitely! Are these the things you're thinking of?
    1. Bootstrap a typescript app with `yarn d2 app scripts init --typescript my-app-name`
    2. TypeScript is supported out of the box with Vite [Is this new, compared to CRA?]
    3. [Upcoming] Use `@dhis2/cli-style` runs type checking
4. Sure, we could merge this to master, but I'm kinda hoping the rest of alpha will be in master soon too haha

Reply:
TS was also supported in CRA, but I think it's still worth it to mention it's supported in Vite .. we can also say that we had other improvements for types like adding types to UI library, and to-come-soon types for the API generated out of OpenAPI specs

Reply:
Added some text in the new & upcoming changes sections 👍